test/linter-test.js

```diff
@@ -250,6 +250,21 @@ describe('linter', () => {
             + 'not available for piped-in code.');
         });
 
+        it('does not fail with --print-config and a filename', async () => {
+          const configInString = (value) => {
+            try {
+              return !!JSON.parse(value).rules;
+            } catch (ex) {
+              return false;
+            }
+          };
+          const args = ['--print-config', fixture_warn];
+
+          await linter.invoke(dir, args, '', 0, callback);
+
+          assert.calledOnceWith(callback, null, sinon.match(configInString));
```

Comment:
Instead of a custom matcher, you could use the `matchJson` assertion:

```js
assert.matchJson(callback.firstCall.args[1], {
  rules: match.defined
});
```

See https://sinonjs.github.io/referee/#matchjson

Reply:
oh thanks!  I'm not very familiar with sinon.  let me make these updates.